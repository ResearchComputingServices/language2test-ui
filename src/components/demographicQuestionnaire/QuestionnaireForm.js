import React from 'react';
import Paper from '@material-ui/core/Paper';
import PropTypes from 'prop-types';
import Form from '../form';

const QuestionnaireForm = ({
    data,
    layout,
    dynamicData,
    dynamicLayout,
    onSubmit,
    controls,
}) => (
    <div className='demographic-questionnaire-container'>
        <div className='text-center'>
            <h1 className='mb-3'>Please fill in the necessary Demographic Information below.</h1>
        </div>
        <Paper className='demographic-questionnaire-form-container'>
            <div className='demographic-questionnaire-form'>
                <Form
                    buttons={[{
                        title: 'Save',
                        handler: onSubmit,
                    }]}
                    controls={controls}
                    data={data}
                    layout={layout}
                    sections={[{
                        layout: dynamicLayout,
                        data: dynamicData,
                    }]}
                />
            </div>
        </Paper>
    </div>
);

QuestionnaireForm.propTypes = {
    data: PropTypes.object.isRequired,
    layout: PropTypes.array.isRequired,
    dynamicData: PropTypes.object.isRequired,
    dynamicLayout: PropTypes.array.isRequired,
    onSubmit: PropTypes.func.isRequired,
    controls: PropTypes.object.isRequired,
};

export default QuestionnaireForm;
