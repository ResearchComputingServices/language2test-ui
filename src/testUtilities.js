/* eslint-disable import/prefer-default-export */
/* eslint-disable prefer-const */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-children-prop */
import React from 'react';
import { shallow, mount as mountComponent } from 'enzyme';
import { Router } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import _ from 'lodash';
import historyService from './services/HistoryService';
import { reducers } from './redux/slices';

const mockStore = configureMockStore({ reducers });

function WrapperComponent({ propsFunc, children, Component }) {
    const props = propsFunc();
    return (
        <Component
            children={children}
            {...props}
        />
    );
}

export function createTest(Component, opts = {}, callback) {
    let { props, children, mount, state } = opts;
    if (_.isNil(props)) props = {};
    if (_.isNil(children)) children = null;
    if (mount) {
        return () => {
            state = state || {};
            const mockMath = Object.create(global.Math);
            mockMath.random = () => 0.5;
            global.Math = mockMath;
            const mountedTree = mountComponent((
                <Router history={historyService.getHistory()}>
                    <Provider store={mockStore({ ...state })}>
                        <Component {...props} >
                            {children}
                        </Component>
                    </Provider>
                </Router>));
            _.isFunction(callback) && callback(mountedTree);
        };
    }
    if (_.isFunction(props)) {
        return () => {
            const shallowTree = shallow(
                <WrapperComponent
                    Component={Component}
                    propsFunc={props}
                >
                    {children}
                </WrapperComponent>,
            ).dive();
            _.isFunction(callback) && callback(shallowTree);
        };
    }
    return () => {
        const shallowTree = shallow((
            <Component {...props} >
                {children}
            </Component>
        ));
        _.isFunction(callback) && callback(shallowTree);
    };
}

export function createSnapshotTest(Component, opts = {}) {
    return createTest(Component, opts, tree => expect(toJSON(tree)).toMatchSnapshot());
}
