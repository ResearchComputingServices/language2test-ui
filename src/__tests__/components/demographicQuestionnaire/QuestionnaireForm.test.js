import React from 'react';
import QuestionnaireForm from '../../../components/demographicQuestionnaire/QuestionnaireForm';
import sampleLayout from '../../../config/formLayouts/demographicQuestionnaireFormLayout';
import { createSnapshotTest } from '../../../testUtilities';

describe('QuestionnaireForm', () => {
    it('should match snapshot', createSnapshotTest(QuestionnaireForm, {
        props: {
            data: {},
            dynamicData: {},
            dynamicLayout: [],
            layout: sampleLayout,
            onSubmit: jest.fn(),
            controls: {},
        },
        children: <div>Test Child</div>,
    }));
});
