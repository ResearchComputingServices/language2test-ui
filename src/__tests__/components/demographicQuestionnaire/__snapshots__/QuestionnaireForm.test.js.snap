// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireForm should match snapshot 1`] = `
<div
  className="demographic-questionnaire-container"
>
  <div
    className="text-center"
  >
    <h1
      className="mb-3"
    >
      Please fill in the necessary Demographic Information below.
    </h1>
  </div>
  <WithStyles(ForwardRef(Paper))
    className="demographic-questionnaire-form-container"
    elevation={2}
  >
    <div
      className="demographic-questionnaire-form"
    >
      <Form
        buttons={
          Array [
            Object {
              "handler": [MockFunction],
              "title": "Save",
            },
          ]
        }
        controls={Object {}}
        data={Object {}}
        layout={
          Array [
            Object {
              "disabled": true,
              "field": "name",
              "required": true,
              "title": "Username",
              "type": "text",
            },
            Object {
              "disabled": true,
              "field": "firstName",
              "required": true,
              "title": "First Name",
              "type": "text",
            },
            Object {
              "disabled": true,
              "field": "lastName",
              "required": true,
              "title": "Last Name",
              "type": "text",
            },
          ]
        }
        sections={
          Array [
            Object {
              "data": Object {},
              "layout": Array [],
            },
          ]
        }
      />
    </div>
  </WithStyles(ForwardRef(Paper))>
</div>
`;
